ibis/common/validators.py

```diff
@@ -8,18 +8,26 @@
 import toolz
 
 from ibis.common.exceptions import IbisTypeError
-from ibis.util import flatten_iterable, is_function, is_iterable
+from ibis.util import DotDict, flatten_iterable, is_function, is_iterable
 
 EMPTY = inspect.Parameter.empty  # marker for missing argument
+VAR_POSITIONAL = inspect.Parameter.VAR_POSITIONAL
+POSITIONAL_OR_KEYWORD = inspect.Parameter.POSITIONAL_OR_KEYWORD
 
 
-class Validator(Callable):
+# TODO(kszucs): rename Validator to Annotation
+class Annotation(Callable):
```

Comment:
I'd prefer to rename this module to `annotations.py`

Reply:
Sounds good to me!

Reply:
Renamed in #4552